github/git_commits.go

```diff
@@ -133,3 +150,51 @@ func (s *GitService) CreateCommit(ctx context.Context, owner string, repo string
 
 	return c, resp, nil
 }
+
+func createSignature(signingKey *openpgp.Entity, commit *createCommit) (string, error) {
+	if signingKey == nil || commit == nil {
+		return "", errors.New("createSignature: invalid parameters")
+	}
+
+	message, err := createSignatureMessage(commit)
+	if err != nil {
+		return "", err
+	}
+
+	writer := new(bytes.Buffer)
+	reader := bytes.NewReader([]byte(message))
+	if err := openpgp.ArmoredDetachSign(writer, signingKey, reader, nil); err != nil {
+		return "", err
+	}
+
+	return writer.String(), nil
+}
+
+func createSignatureMessage(commit *createCommit) (string, error) {
+	if commit == nil || commit.Message == nil || commit.Author == nil {
```

Comment:
It just dawned on me that `commit.Message == nil` should be `commit.GetMessage() == ""` because we also care that there is no message, and this helper function does the extra work for us.

I apologize for not catching this one before.

Reply:
Remember there are no getters for type `createCommit`

But will check for `nil` and empty string.